barretenberg/cpp/src/barretenberg/vm2/simulation/execution.cpp

```diff
@@ -84,15 +87,10 @@ void Execution::jumpi(ContextInterface& context, MemoryAddress cond_addr, uint32
     }
 }
 
-ExecutionResult Execution::execute(AztecAddress contract_address,
-                                   std::span<const FF> calldata,
-                                   AztecAddress msg_sender,
-                                   bool is_static)
+// This context interface is an top-level enqueued one
```

Comment:
Comment is a bit confusing

Reply:
Ah yeah - strictly speaking this `execute` should really only take in an `EnqueuedCallContext` and never a `NestedCallContext`. But I wasn't sure if removing the polymorphism would be appropriate right now.

We might have a `(Enqueued)/(Nested)ContextInterface` in the end if they diverge significantly which is what i would then put here.

Reply:
Ha! If this requires you to know the specific kind of context then usually either this is wrong or the interface is wrong 😆 but we can see later